client/allocrunner/networking_bridge_linux.go

```diff
@@ -134,8 +138,44 @@ func (b *bridgeNetworkConfigurator) Teardown(ctx context.Context, alloc *structs
 	return b.cni.Teardown(ctx, alloc, spec)
 }
 
-func buildNomadBridgeNetConfig(bridgeName, subnet string) []byte {
-	return []byte(fmt.Sprintf(nomadCNIConfigTemplate, bridgeName, subnet, cniAdminChainName))
+func buildNomadBridgeNetConfig(b bridgeNetworkConfigurator) []byte {
+	// Parse the internal template that generates the Nomad bridge interface.
+	// The template's parsablity is tested in networking_bridge_linux_test.go
+	// so any template issues should be caught by the test rather than at this
+	// point.
+	tmpl, err := template.New("cniConf").Parse(nomadCNIConfigTemplate)
+	if err != nil {
+		// Panic on error for catching issues in testing
+		panic(err)
+	}
+
+	// TODO: Consider exporting these directly from bridgeNetworkConfigurator
+	// so they aren't repeated in the input struct
+	type templInput struct {
+		AllocSubnet       string
+		BridgeName        string
+		HairpinMode       bool
+		CNIAdminChainName string
+	}
+
+	tIn := templInput{
+		AllocSubnet:       b.allocSubnet,
+		BridgeName:        b.bridgeName,
+		HairpinMode:       b.hairpinMode,
+		CNIAdminChainName: cniAdminChainName,
+	}
+
+	var out bytes.Buffer
+
+	// Since the input object provides all of the variable elements used in the
+	// template and out is a buffer (rather than a closable writer) the template
+	// execution should not error.
+	err = tmpl.Execute(&out, tIn)
+	if err != nil {
+		// Panic on error for catching issues in testing
+		panic(err)
+	}
```

Comment:
The docs are not very clear in all possible error modes here, so I think it may be better to return `err` and let the allocrunner handle that? It may be a transient issue that a retry would fix.